src/com/yetanalytics/lrs_admin_ui/input.cljs

```diff
@@ -0,0 +1,32 @@
+(ns com.yetanalytics.lrs-admin-ui.input
+  (:require [cljs.spec.alpha :as s :include-macros true]
+            [com.yetanalytics.lrs-admin-ui.functions.password :as pass]))
+
+;; Minimum lengths
+(def p-min-len 10)
+(def u-min-len 7)
+
+(def digit-set (set pass/digit-chars))
+(def upper-set (set pass/upper-chars))
+(def lower-set (set pass/lower-chars))
+(def special-set (set pass/special-chars))
+
+(s/def :valid-account/password
+  (s/and string?
+         #(>= (count %) p-min-len)
+         #(let [pass-set   (set %)
+                has-chars? (partial some pass-set)]
+            (and
+             (has-chars? digit-set)
+             (has-chars? upper-set)
+             (has-chars? lower-set)
+             (has-chars? special-set)))))
+
+(s/def :valid-account/username
+  (s/and string?
+         #(> (count %) u-min-len)
```

Comment:
Username is "strict greater than `u-min-len`" but password is "greater than equal to `p-min-len` here."

Reply:
ah good one! i updated one after realizing the mistake but forgot the other.